examples/double_pendulum.jl

```diff
@@ -0,0 +1,119 @@
+using DifferentialEquations, Plots
+
+#Reference: https://diego.assencio.com/?index=e5ac36fcb129ce95a61f8e8ce0572dbf
+
+
+#Solving the double pendulum with a traditional ODE method
+#==========================================================#
+function doublependulum(du, u, params, t)
+    l1 = params[1]
+    l2 = params[2]
+    m1 = params[3]
+    m2 = params[4]
+    g  = params[5]
+
+    p1 = u[1]
+    p2 = u[2]
+    θ1 = u[3]
+    θ2 = u[4]
+
+    h1 = p1*p2*sin(θ1-θ2)/(l1*l2*(m1+m2*sin(θ1-θ2)^2))
+    h2 = (m2*l2^2*p1^2 + (m1 + m2) * l1^2 * p2^2 - 2*m2*l1*l2*p1*p2*cos(θ1-θ2))/(2 * l1^2 * l2^2 *(m1 + m2*sin(θ1-θ2)^2)^2)
+
+    d_p1 = -(m1 + m2)* g*l1*sin(θ1) - h1 + h2*sin(2*(θ1-θ2))
+    d_θ1 = (l2*p1 - l1*p2*cos(θ1-θ2))/(l1^2 * l2*(m1 + m2*sin(θ1 - θ2)^2))
+    d_p2 = -m2*g*l2*sin(θ2) + h1 - h2*sin(2*(θ1-θ2))
+    d_θ2 = (-m2*l2*p1*cos(θ1 - θ2) + (m1 + m2)*l1*p2) / (m2*l1*l2^2*(m1 + m2*sin(θ1-θ2)^2))
+
+
+    du .= [d_p1, d_p2, d_θ1, d_θ2]
+    return nothing
+end
+
+l1 = 1. #length of pendulum1
+l2 = 2. #length of pendulum2
+m1 = 1. #mass of pendulum1
+m2 = 1. #mass of pendulum2
+g = 9.81 #gravity
+
+params = [l1, l2, m1, m2, g]
+times = (0., 25.)
+u0 = [1.,1.,1.,1.]
+prob = ODEProblem(doublependulum, u0, times, params)
+sol1 = solve(prob, SSPRK83(), dt = .05)
+
+#plot solution
+plot(sol1, vars=1, xlim=(0,20), label="Momentum1")
+plot!(sol1, vars=2, xlim=(0,20), label="Momentum2")
+plot!(sol1, vars=3, xlim=(0,20), label="theta1")
+plot!(sol1, vars=4, xlim=(0,20), label="theta2")
+#==========================================================#
+
+
+#Now with HamiltonianProblem()
+#==========================================================#
+function H(p, θ, params)
+    l1 = params[1]
+    l2 = params[2]
+    m1 = params[3]
+    m2 = params[4]
+    g  = params[5]
+
+    return  (m2*l2^2*p[1]^2 + (m1+m2)*l1^2*p[2]^2 - 2*m2*l1*l2*p[1]*p[2]*cos(θ[1]-θ[2]) ) /
+        (2*m2*l1^2*l2^2*(m1+m2*sin(θ[1]-θ[2])^2)) -
+        (m1+m2)*g*l1*cos(θ[1]) - m2*g*l2*cos(θ[2])
+end
+
+l1 = 1. #length of pendulum1
+l2 = 2. #length of pendulum2
+m1 = 1. #mass of pendulum1
+m2 = 1. #mass of pendulum2
+g = 9.81 #gravity
+
+params = [l1, l2, m1, m2, g]
+q0 = [1.0,1.0]
+p0 = [1.0,1.0]
+times = (0.,25.)
+prob = HamiltonianProblem(H, q0, p0, times, p=params)
+sol2 = solve(prob, SSPRK83(), dt = .05)
```

Comment:
why an SSP method?

Reply:
It has high order accuracy and a constant time step. Are you recommending a symplectic solver would go better here?

Reply:
Yes. SSP methods are a bad choice here since they take a low of stages for what's quite low accuracy. SSP methods are built so that when they are applied to PDEs with CFL constraints they increase the possible stepsize, since it's CFL-dominated and not error dominated. However, those extra relations to make it strong stability preserving (SSP) means that they have to have more stages than a regular method to get the same order. So if you're not using the SSP-ness of the method, it's much less efficient than other choices. A symplectic integrator or adaptive Runge-Kutta Nystrom method are a much better choice.